tools/file_packager.py

```diff
@@ -290,18 +290,32 @@ def generate_object_file(data_files):
 
     if options.wasm64:
       align = 3
+      ptr_type = 'i64'
     else:
       align = 2
-    out.write(dedent('''
+      ptr_type = 'i32'
+    out.write(dedent(f'''
+      .functype _emscripten_fs_load_embedded_files ({ptr_type}) -> ()
+      .section .text,"",@
+      init_file_data:
+        .functype init_file_data () -> ()
+        global.get __emscripten_embedded_file_data@GOT
+        call _emscripten_fs_load_embedded_files
+        end_function
+
+      .section  .init_array.100,"",@
+      .p2align  2
+      .int32 init_file_data
```

Comment:
How do we ensure that this global ctor runs before user ones? Is it the `100` constant in there perhaps?

Reply:
Ah yes, sorry I forgot to allocate a specific ordering.   Fixed.